### ISSUE TITLE
Player attach and despawn bug

### ISSUE THREAD

**Issue by xxshady** [altmp/altv-issues#1456]
### Description of the problem

Player despawn and spawn methods together can cause sync problems. See repro steps 

### Reproduction steps

```js
import * as alt from "alt-server";

const [firstPlayer, secondPlayer] = alt.Player.all;

player.attachTo(secondPlayer, 0, 0, new alt.Vector3(0, 0, 1.0), new alt.Vector3(0, 0, 1.0), true, false);

// despawn doesnt call player.detach() for the second player somewhy
alt.setTimeout(() => firstPlayer.despawn(), 1000);
alt.setTimeout(() => firstPlayer.spawn(firstPlayer.pos), 2000);

```

result:

![image](https://user-images.githubusercontent.com/54737754/166834752-7ea63a3c-99f2-483b-9eda-c19bad991a94.png)

### Expected behaviour

should despawn call detach or should we describe this behaviour in the docs?

### Operating system

Windows 11

### Version

dev/10.0-dev20

**Comment by YannBcf** [comment]
Confirmed on 10.0-dev21

**Comment by FabianTerhorst** [comment]
Should be fixed in dev/15.0-dev341
